weatherman.py

```diff
@@ -0,0 +1,208 @@
+import argparse
+import calendar
+import csv
+import os
+
+class TemperatureTracker:
+
+    def __init__(self):       
+        self.max_temperature = float('-inf')
+        self.max_temperature_date = ""
+        self.min_temperature = float('inf')
+        self.min_temperature_date = ""
+        self.max_humidity = float('-inf')
+        self.max_humidity_date = ""
+    
+    def track_max_temperature(self, temperature, date): 
+        if temperature != "":
+            self.max_temperature = max(self.max_temperature, int(temperature))
+            if self.max_temperature == int(temperature): self.max_temperature_date = date 
+        
+    def track_min_temperature(self, temperature, date): 
+        if temperature != "":
+            self.min_temperature = min(self.min_temperature, int(temperature))
+            if self.min_temperature == int(temperature): self.min_temperature_date = date
+
+    def track_max_humidity(self, humidity, date):
+        if humidity != "":
+            self.max_humidity = max(self.max_humidity, int(humidity))
+            if self.max_humidity == int(humidity): self.max_humidity_date = date
+
+    def print_temperatures_year(self):
+        print(f"highest temperature is: {self.max_temperature} C on {self.max_temperature_date}")
+        print(f"lowest temperature is: {self.min_temperature} C on {self.min_temperature_date}")
+        print(f"highest humidity percentage is: {self.max_humidity} % on {self.max_humidity_date}")
+
+
+class WeatherFilesHandler:
+
+    def __open_csv_file(self, file_name):    
+        csv_file = open(file_name) 
+        return csv_file
+
+    def __read_weather_data_from_csv(self, csv_file, file_name):
+        weather_data = list(csv.DictReader(csv_file, skipinitialspace=True))
+        if len(weather_data) != 0: return weather_data
+        print(f"No Record Found against file : {file_name}")
+    
+    def return_weather_data(self, file_name):
+        csv_file = self.__open_csv_file(file_name)
+        weather_data = self.__read_weather_data_from_csv(csv_file, file_name)
+        return weather_data
+
+    def return_all_file_names_list_for_year(self, year):
+        file_names=[]
+        flag = False
+        for file_name in os.listdir('.'):
+            if year in file_name:
+                file_names.append(file_name)
+                flag = True
+        return file_names if flag else 0
```

Comment:
Return format should be consistent. In one case we are returning boolean and in one case its integer.